src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -405,20 +405,21 @@ public function send(OutgoingMessage $message)
 	    if (@fwrite($this->_socket, $messageToSend) < $length) {
     	    throw new ClientException('Could not send message');
 	    }
-	    $read = 0;
-	    while($read <= $this->_rTimeout) {
-	        $this->process();
-	        $response = $this->getRelated($message);
-	        if ($response != false) {
-	            $this->_lastActionId = false;
-	            return $response;
-	        }
-	        usleep(1000); // 1ms delay
-	        if ($this->_rTimeout > 0) {
-	            $read++;
-	        }
-	    }
-	    throw new ClientException('Read timeout');
+	    while (1) {
+	    	@stream_set_timeout($this->_socket, $this->_rTimeout ? $this->_rTimeout : 1);
```

Comment:
Hi! First of all: nice contribution, thanks!

I see in both cases you use the error suppression operator on this line and on line 411. In general this indicates that the associated problems are handled elsewhere. Is that the case here?

If not, could you think about which behaviour will change when one of these fail and find a solution for that? Having a random error throwing up in time-sensitive code that is used in a production environment is not fun at all :smile: